### PR TITLE
Reintroducing parse_a_sizes along with tests

### PR DESCRIPTION
**Pull request by paavininanda**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #__ (github issue number if applicable).

- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21178)

### CODE REVIEW

File: components/script/dom/htmlimageelement.rs [hunk 1]
```diff
@@ -1010,6 +1017,36 @@ pub fn collect_sequence_characters<F>(s: &str, predicate: F) -> (&str, &str)
...
+    let mut source_size_list = SourceSizeList::parse(&context, &mut parser);
+    if source_size_list.source_sizes.len() == 0  {
+        source_size_list = match width {
+            Some(w) => SourceSizeList {
+                source_sizes: vec![],
+                value: Some(Length::from_px(w as f32))
+            },
+            None => SourceSizeList {
+                source_sizes: vec![],
+                value: Some(Length::NoCalc(NoCalcLength::ViewportPercentage(ViewportPercentageLength::Vw(100.))))
```
Comment (emilio) [inline components/script/dom/htmlimageelement.rs]: Also note that this is not needed, `SourceSizeList::evaluate` already handles this correctly.

File: tests/unit/script/htmlimageelement.rs [hunk 2]
```diff
@@ -2,7 +2,90 @@
...
+use script::test::sizes::parse_a_sizes_attribute;
 use script::test::srcset::{Descriptor, ImageSource, parse_a_srcset_attribute};
+use style::media_queries::{MediaCondition, MediaFeatureExpression};
+use style::servo::media_queries::{ExpressionKind, Range};
+use style::values::specified::{Length, NoCalcLength, AbsoluteLength};
+use style::values::specified::{source_size_list::SourceSizeList, source_size_list::SourceSize};
+
+
+#[test]
+fn no_default_provided() {
```
Comment (emilio) [inline tests/unit/script/htmlimageelement.rs]: I think I'd prefer to remove the unit tests and make the `SourceSizeList` and `SourceSize` fields private again, given there's a few tests that check all this in WPT: https://wpt.fyi/results/html/semantics/embedded-content/the-img-element/sizes

Comment (jdm) [reply tests/unit/script/htmlimageelement.rs]: I would prefer to retain the unit tests for the time being until we finish implementing all the pieces that are required in order for the WPT tests to actually work in Servo.

Comment (emilio) [reply tests/unit/script/htmlimageelement.rs]: That means that when I change SourceSizeList in Gecko I need to update all the tests, which is cumbersome given this is tested already there... :/

Comment (jdm) [reply tests/unit/script/htmlimageelement.rs]: Ok, that's a reasonable argument against these changes. @paavininanda Let's undo the changes that were made to support re-adding the unit tests, and remove the unit tests. This PR can just focus on adding the parse_a_sizes_attribute function without using it anywhere.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by jdm** [comment]
Since this is part of #21181 which makes use of it, let's just wait for that PR to merge.

**Pull request closed without merging**